app_dart/lib/src/service/firestore.dart

```diff
@@ -27,4 +32,67 @@ class FirestoreService {
     );
     return FirestoreApi(client).projects.databases.documents;
   }
+
+  /// Writes [writes] to Firestore within a transaction.
+  ///
+  /// This is an atomic operation: either all writes succeed or all writes fail.
+  Future<CommitResponse> writeViaTransaction(List<Write> writes) async {
+    final ProjectsDatabasesDocumentsResource databasesDocumentsResource = await documentResource();
+    final BeginTransactionRequest beginTransactionRequest =
+        BeginTransactionRequest(options: TransactionOptions(readWrite: ReadWrite()));
+    final BeginTransactionResponse beginTransactionResponse =
+        await databasesDocumentsResource.beginTransaction(beginTransactionRequest, kDatabase);
+    final CommitRequest commitRequest =
```

Comment:
If commit fails do we need a rollback?

Reply:
Transaction acts as an atomic operation. If it fails, everything will auto roll back.